src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -521,9 +542,83 @@ else if (job.getTree().hasLeaves() && job.getTree().isSlimeTree())
             }
             job.getTree().pollNextLeaf();
         }
+        else if (job.getTree().hasLeaves() && job.getTree().isNetherTree())
+        {
+            final BlockPos leaf = job.getTree().peekNextLeaf();
+            if (!mineBlock(leaf, workFrom))
+            {
+                return getState();
+            }
+            job.getTree().pollNextLeaf();
+        }
         return getState();
     }
 
+    /**
+     * Bonemeal fungi in netherTrees
+     */
+
+    private IAIState bonemealFungi()
+    {
+        for (final BlockPos pos : getNetherTrees())
+        {
+            final BlockState blockState = world.getBlockState(pos);
+            final Block block = blockState.getBlock();
+            if (block == Blocks.CRIMSON_FUNGUS || block == Blocks.WARPED_FUNGUS)
+            {
+                final int threshold = worker.getCitizenData().getCitizenSkillHandler().getLevel(getOwnBuilding().getPrimarySkill()) * FUNGI_MODIFIER;
+                final int rand = new Random().nextInt(100);
+                if (rand < threshold)
+                {
+                    final IGrowable growable = (IGrowable) block;
+                    if (growable.canGrow(world, pos, blockState, world.isRemote)) {
+                        if (!world.isRemote) {
+                            if (growable.canUseBonemeal(world, world.rand, pos, blockState)) {
+                                growable.grow((ServerWorld) world, world.rand, pos, blockState);
+                            }
+                        }
+                    }
+                }
+
+            }
+            else
+            {
+                removeNetherTree(pos);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Drop fungus instead of wart block
+     *
+     * @param drops the drops.
+     * @return the list of additional drops.
+     */
+    @Override
+    protected List<ItemStack> increaseBlockDrops(final List<ItemStack> drops)
+    {
+        final List<ItemStack> newDrops = new ArrayList<>();
+        for (final ItemStack stack : drops)
+        {
+            boolean modified = false;
+            if (new Random().nextInt(10) > 8) {
```

Comment:
avoid new random here too

Reply:
and I think an 20% chance is a bit much maybe? think it should be similar to what vanilla's sapling chances from leaves, those are 5%

Reply:
there is a worker.getRandom() or getRand()